### ISSUE TITLE
When training crossencoder, encounter the problem that the model is easy to fall into the local optimum

### ISSUE THREAD

**Issue by zbrnwpu** [UKPLab/sentence-transformers#1077]
hello，@nreimers 
1. There is no difference between model.zero_grad() and optimizer.zero_grad() when training the crossencoder model? Can you explain it in detail?
2. As shown in the figure, the model gradient is set to 0 after the evaluation. Here I feel that the number of steps in the evaluation is the number of iterations  in 1 epoch.Will the model easily fall into the local optimum?
![image](https://user-images.githubusercontent.com/34335281/126424569-7629c6b4-23ea-4995-be24-e6c1e66377de.png)

**Comment by nreimers** [comment]
1) Both operations do the same
2) Not sure what you mean. No, it will not fall into a local optimum. The code there just ensures that we do not keep any gradients from the evaluation (which should not be the case anyway).

**Comment by zbrnwpu** [comment]
> 1. 两个操作都是一样的
> 2. 不明白你的意思。不，它不会陷入局部最优。那里的代码只是确保我们不会在评估中保留任何梯度（无论如何都不应该是这种情况）。

Thank you, do you know how to check the current learning rate during training?
